code/game/objects/items/devices/multitool.dm

```diff
@@ -98,30 +97,89 @@
 
 // Syndicate device disguised as a multitool; it will turn red when an AI camera is nearby.
 
-
 /obj/item/multitool/ai_detect
 	var/track_cooldown = 0
 	var/track_delay = 10 //How often it checks for proximity
 	var/detect_state = PROXIMITY_NONE
 	var/rangealert = 8	//Glows red when inside
 	var/rangewarning = 20 //Glows yellow when inside
+	var/hud_type = DATA_HUD_AI_DETECT
+	var/hud_on = FALSE
+	var/mob/camera/aiEye/remote/ai_detector/eye
+	var/datum/action/item_action/toggle_multitool/toggle_action
 
-/obj/item/multitool/ai_detect/New()
-	..()
+/obj/item/multitool/ai_detect/Initialize()
+	. = ..()
 	START_PROCESSING(SSobj, src)
+	eye = new /mob/camera/aiEye/remote/ai_detector()
+	toggle_action = new /datum/action/item_action/toggle_multitool(src)
 
 /obj/item/multitool/ai_detect/Destroy()
 	STOP_PROCESSING(SSobj, src)
+	if(hud_on && ismob(loc))
+		remove_hud(loc)
+	QDEL_NULL(toggle_action)
+	QDEL_NULL(eye)
 	return ..()
 
+/obj/item/multitool/ai_detect/ui_action_click()

```

Comment:
We don't proliferate ui_action_click anymore. Use a custom action datum

Reply:
I did use a custom action datum, this is here to suppress the default behavior of ui_action_click().